### PR TITLE
cli: add --skip-lint option

### LINKED ISSUE #1469: cli: provide --skip-lint option to anchor build
https://github.com/project-serum/anchor/pull/1452 introduced linting. We should add a `anchor build --skip-lint` flag so that devs can rapidly prototype.

### PR DESCRIPTION
**Pull request by paul-schaaf**
closes #1469 

submodule PRs:
- https://github.com/armaniferrante/auction-house/pull/7
- https://github.com/project-serum/stake/pull/24
- https://github.com/project-serum/swap/pull/29

### CODE REVIEW

File: cli/src/lib.rs [hunk 1]
```diff
@@ -1392,7 +1433,7 @@ fn extract_idl(cfg: &WithPath<Config>, file: &str) -> Result<Option<Idl>> {
         &*file,
         cargo.version(),
         cfg.features.seeds,
-        cfg.features.safety_checks,
+        !skip_lint && cfg.features.safety_checks,
```
Comment (armaniferrante) [inline cli/src/lib.rs:1436]: should we remove the safety_checks ffeature?

Comment (paul-schaaf) [reply cli/src/lib.rs:1436]: yea it's pretty much redundant now isnt it

Comment (armaniferrante) [reply cli/src/lib.rs:1436]: yup

Comment (tomlinton) [reply cli/src/lib.rs:1436]: I'm happy to do the work on that in another PR since I put the flag everywhere 😬

Comment (paul-schaaf) [reply cli/src/lib.rs]: ah just saw this. already removed it here

**Comment by armaniferrante** [comment]
Submodules merged. CFO test failing.

**Pull request merged**